modules/core/server/services/influx.server.service.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+var path = require('path'),
+    influx = require('influx'),
+    config = require(path.resolve('./config/config'));
+
+/**
+ * Get InfluxDB Client
+ */
+exports.getClient = function(callback) {
+  if (!config.influxdb ||
+      !config.influxdb.enabled ||
+      !config.influxdb.options ||
+      !config.influxdb.options.host ||
+      !config.influxdb.options.database) {
+    return callback(new Error('No InfluxDB configured.'));
+  }
+  callback(null, influx(config.influxdb.options));
+};
+
+/**
+ * Write point to InfluxDB
+ *
+ * `values` can be either an object or a single value. For the latter the columname is set to value.
+ * You can set the time by passing an object propety called time. The time an be either an integer value or a Date object.
+ * When providing a single value, don't forget to adjust the time precision accordingly. The default value is `ms`.
+ */
+exports.writePoint = function(seriesName, values, tags, callback) {
+
+  if (!seriesName || typeof seriesName !== 'string' || seriesName.length === 0) {
+    return callback(new Error('InfluxDB Service: no `seriesName` defined.'));
+  }
+
+  if (values === undefined || values === null) {
+    return callback(new Error('InfluxDB Service: no `values` defined.'));
+  }
+
+  if (!tags || typeof tags !== 'object') {
+    return callback(new Error('InfluxDB Service: no `tags` defined.'));
+  }
+
+  // Turn `values` into object so that we can add time if it's missing
+  if (typeof values !== 'object') {
+    values = {
+      value: values
+    };
+  }
```

Comment:
Great that you're testing `values`. In this case, if `values` is an `array`, a `Date` or a `function`, then it will be passed along to influxdb. Are all of those types valid? I'm definitely not 100% sure, but I'd guess if `values` is not an object, it should probably not be a `function` and maybe should only be one of a few types. Not strictly necessary to check so thoroughly, but also no harm in it.